Source/WebCore/Modules/mediastream/RTCPeerConnection.cpp

```diff
@@ -215,6 +215,42 @@ void RTCPeerConnection::createOffer(RTCOfferOptions&& options, Ref<DeferredPromi
         return;
     }
 
+    // https://www.w3.org/TR/webrtc/#legacy-configuration-extensions
+    auto processLegacyOption = [&](auto option, auto trackKind) {
+        if (!option) {
+            for (auto& transceiver : currentTransceivers()) {
+                if (transceiver->stopped())
+                    continue;
+                if (transceiver->sender().trackKind() != trackKind)
+                    continue;
+                if (transceiver->direction() == RTCRtpTransceiverDirection::Sendrecv)
+                    transceiver->setDirection(RTCRtpTransceiverDirection::Sendonly);
+                else if (transceiver->direction() == RTCRtpTransceiverDirection::Recvonly)
+                    transceiver->setDirection(RTCRtpTransceiverDirection::Inactive);
+            }
+            return;
+        }
+
+        for (auto& transceiver : currentTransceivers()) {
+            if (transceiver->stopped())
+                continue;
+            if (transceiver->sender().trackKind() != trackKind)
+                continue;
+            auto direction = transceiver->direction();
+            if (direction == RTCRtpTransceiverDirection::Sendrecv || direction == RTCRtpTransceiverDirection::Recvonly)
+                return;
+        }
+
+        RTCRtpTransceiverInit init { .direction = RTCRtpTransceiverDirection::Recvonly, .streams = { }, .sendEncodings = { } };
+        addTransceiver(trackKind, init);
```

Comment:
This will likely triggers negotiation needed flag. It might be best to pipe it platform specific code that will handle this part properly.

Reply:
Hm yes, this needs some work indeed. I noticed a couple more issues. There's also a couple WPT tests related to these options too. I'll rework this PR. Thanks for the review!